world/combat/combat_functions.py

```diff
@@ -524,3 +545,21 @@ def vigor_check(character, base_stat):
     if character.db.vigor_duration > 0:
         base_stat += 25
     return base_stat
+
+
+def dispel_check(target):
+    # If an Art with Dispel successfully hits or is endured, check for buffs on the target and remove one at random.
+    dispel_targets = []
+    for status_effect, duration in target.db.status_effects.items():
```

Comment:
As we discussed on chat: in might be worth splitting the status effects dictionary into two different dictionaries so you don't have to keep sifting through the single dictionary for buffs/debuffs